pkg/storage/postgres/postgres.go

```diff
@@ -54,63 +52,87 @@ type Datastore struct {
 // Ensures that Datastore implements the OpenFGADatastore interface.
 var _ storage.OpenFGADatastore = (*Datastore)(nil)
 
-// initDB initializes a new postgres database connection.
-func initDB(uri string, username string, password string, cfg *sqlcommon.Config) (*sql.DB, error) {
-	if username != "" || password != "" {
+func parseConfig(uri string, override bool, cfg *sqlcommon.Config) (*pgxpool.Config, error) {
+	c, err := pgxpool.ParseConfig(uri)
+	if err != nil {
+		return nil, fmt.Errorf("pgxpool parse postgres connection uri: %w", err)
+	}
+
+	if override {
 		parsed, err := url.Parse(uri)
 		if err != nil {
-			return nil, fmt.Errorf("parse postgres connection uri: %w", err)
+			return nil, fmt.Errorf("url parse postgres connection uri: %w", err)
 		}
 
-		username := ""
 		if cfg.Username != "" {
-			username = cfg.Username
+			c.ConnConfig.User = cfg.Username
 		} else if parsed.User != nil {
-			username = parsed.User.Username()
+			c.ConnConfig.User = parsed.User.Username()
 		}
 
 		switch {
 		case cfg.Password != "":
-			parsed.User = url.UserPassword(username, cfg.Password)
+			c.ConnConfig.Password = cfg.Password
 		case parsed.User != nil:
 			if password, ok := parsed.User.Password(); ok {
-				parsed.User = url.UserPassword(username, password)
-			} else {
-				parsed.User = url.User(username)
+				c.ConnConfig.Password = password
 			}
-		default:
-			parsed.User = url.User(username)
 		}
+	}
 
-		uri = parsed.String()
+	if cfg.MaxOpenConns != 0 {
+		c.MaxConns = int32(cfg.MaxOpenConns)
 	}
 
-	db, err := sql.Open("pgx", uri)
-	if err != nil {
-		return nil, fmt.Errorf("initialize postgres connection: %w", err)
+	if cfg.MinIdleConns != 0 {
+		c.MinIdleConns = int32(cfg.MinIdleConns)
+	} else if cfg.MaxOpenConns != 0 {
+		// Use some sane value for defaults
+		c.MinIdleConns = int32(cfg.MaxOpenConns)
```

Comment:
With this change we'll end up holding the maximum number of allowed connections open at all times unless the config is updated to set both minimums. I'm not sure that setting it to zero is the correct solution either (since we'll end up churning connections needlessly), but for users who have configured a relatively high maximum it could result in a significant change in the number of open connections under normal usage.

Would it make sense to default these limits to some fraction of the maximum?